src/HexagonOptimize.cpp

```diff
@@ -178,7 +180,14 @@ Expr apply_patterns(Expr x, const vector<Pattern> &patterns, const Target &targe
     vector<Expr> matches;
     for (const Pattern &p : patterns) {
 
-        if ((p.flags & (Pattern::v62)) && !target.has_feature(Target::HVX_v62))
+        if ((p.flags & (Pattern::v62orLater)) &&
+            !target.features_any_of({Target::HVX_v62, Target::HVX_v65, Target::HVX_v66}))
```

Comment:
One way to fix this would be that when constructing a Target (parsing or otherwise), if Target::HVX_vX is present, also add Target::HVX_vY where Y < X. That way, any logic like this can assume that HVX_vX implies HVX_vY is present.